### ISSUE TITLE
v1.3.1 breaks custom post type categories

### ISSUE THREAD

**Issue by jooola** [wp-graphql/wp-graphql#1802]
I have a custom post type registered in my wordpress, and since the v1.3.1 upgrade, I cannot query categories on this custom post type. I don't think it was a bug being fixed in this release.

The field simply  doesn't exist anymore.

 "gatsby-source-wordpress": "^5.1.0"
WP-Gatsby Version 1.0.6
WP-Graphql Version 1.3.1

I query this data from gatsby, and I checked in the WP graphql IDE if it was Gatsby or WP-graphql related. This really comes from the upgrade from wp-graphql 1.2.6 to 1.3.1

Here is how I register my custom post type:
```php
add_action('init', function () {
    register_post_type('shows', [
        'label' => __('Émission', 'aro_theme'),
        'description' => __('Émissions', 'aro_theme'),
        'labels' => [
            'name' => _x('Émissions', 'Post Type General Name', 'aro_theme'),
            'singular_name' => _x('Émission', 'Post Type Singular Name', 'aro_theme'),
            'menu_name' => __('Émissions', 'aro_theme'),
            'name_admin_bar' => __('Émissions', 'aro_theme'),
        ],
        'supports' => ['title', 'editor', 'author', 'comments', 'revisions', 'excerpt'],
        'taxonomies' => ['category'],
        'hierarchical' => false,
        'public' => true,
        'show_ui' => true,
        'show_in_menu' => true,
        'menu_position' => 5,
        'menu_icon' => 'dashicons-format-audio',
        'show_in_admin_bar' => true,
        'show_in_nav_menus' => true,
        'can_export' => true,
        'has_archive' => true,
        'exclude_from_search' => false,
        'publicly_queryable' => true,
        'capability_type' => 'page',

        'rewrite' => [
            'slug' => 'emissions',
            'with_front' => false,
        ],
        'cptp_permalink_structure' => '/%category%/%postname%/',

        'show_in_graphql' => true,
        'graphql_single_name' => 'show',
        'graphql_plural_name' => 'shows',
    ]);
}, 0);
```

Here is one of my graphql query:
```
  query($id: String!) {
    wpShow(id: { eq: $id }) {
      uri
      title
      content
      date
      categories {
        nodes {
          uri
          name
        }
      }
      author {
        node {
          uri
          name
        }
      }
      meta {
        file {
          mediaItemUrl
        }
      }
    }
  }
```

And some logs from Gatsby:

```
File: src/templates/show.js:63:7

failed extract queries from components - 0.084s
success write out requires - 0.001s

 ERROR #98123  WEBPACK

Generating development JavaScript bundle failed


/home/jo/git/git.oolsa.net/ext/aro/website/src/templates/author.js
  75:9  error  Cannot query field "categories" on type "WpShow"  graphql/template-strings

/home/jo/git/git.oolsa.net/ext/aro/website/src/templates/show.js
  63:7  error  Cannot query field "categories" on type "WpShow"  graphql/template-strings

✖ 2 problems (2 errors, 0 warnings)


failed Re-building development bundle - 0.248s
ERROR in 
/home/jo/git/git.oolsa.net/ext/aro/website/src/templates/author.js
  75:9  error  Cannot query field "categories" on type "WpShow"  graphql/template-strings

/home/jo/git/git.oolsa.net/ext/aro/website/src/templates/show.js
  63:7  error  Cannot query field "categories" on type "WpShow"  graphql/template-strings

✖ 2 problems (2 errors, 0 warnings)


webpack compiled with 1 error

```

**Comment by emajeru** [comment]
I'm experiencing the same issue after updating to 1.3.1 from the auto-updater. This seems to have been introduced in 1.3.0 as I was only able to see categories in my custom post type after downgrading to 1.2.6.

**Comment by jasonbahl** [comment]
Ok, checking this out. I think I might know what the issue is.

**Comment by jasonbahl** [comment]
@jooola @emajeru I just published 1.3.2. 

I had made some changes to how initial post types and taxonomies are added to the Schema to support some test cases that are currently tricky to test. . .it fixed the test environment issues I was having, but sadly broke _real user_ cases. 

I clearly need to make some adjustments to make sure this situation doesn't happen again. Tests shouldn't pass if real users experience something different 😬

**Comment by jooola** [comment]
@jasonbahl v1.3.2 did fixed the bug on my side. Thanks for such a quick fix.

I would go for e2e tests and adding test case for each issue that has been raised. It prevent any regression for any fix, and it build a widespread coverage after some time. But that's just my 2cts

**Comment by emajeru** [comment]
Fixed for me as well in v1.3.2. Thanks @jasonbahl !

**Comment by jasonbahl** [comment]
@jooola ya, the tests we have are pretty much E2E. . .the test environment sets up a WordPress install, activates plugins, flushes permalinks, etc then creates data, queries it, etc. 

In this case, I made a change to the plugin that affected the registration of custom post types, and foolishly made a change to the test code that compensated and allowed the missing post types to still be registered for tests. 🤦‍♂️
